var/spack/repos/builtin/packages/hip/package.py

```diff
@@ -58,6 +58,7 @@ class Hip(CMakePackage):
     # See https://github.com/ROCm-Developer-Tools/HIP/pull/2218
     patch('0003-Improve-compilation-without-git-repo.3.7.0.patch', when='@3.7.0:3.9.0')
     patch('0003-Improve-compilation-without-git-repo.3.10.0.patch', when='@3.10.0:4.0.0')
+    patch('0003-Improve-compilation-without-git-repo.4.1.0.patch', when='@4.1.0')
```

Comment:
Is there any reason this patch is not getting merged upstream?

Reply:
https://github.com/ROCm-Developer-Tools/HIP/pull/2218 I tried. Let me bump that thread again.

Reply:
@haampie I saw that, and I think the patch is appropriate. I was wondering though why it is not accepted, since it seems we have to maintain slightly modified versions of it for different version ranges of the package :slightly_frowning_face:

Reply:
Yeah... another way to fix it is to use git instead of tarballs, but I think they should just support non-git builds.